File: c/use-after-free.c
#include <stdio.h>
#include <stdlib.h>

void use_after_free() {
    int* ptr = (int*)malloc(sizeof(int));
    *ptr = 42;
    
    free(ptr);  // Memory freed
    int value = *ptr;  // UAF! Accessing freed memory.

    printf("Use-After-Free executed\n");
}

int main() {
    use_after_free();
    return 0;
}

File: rust/use-after-free.rs
fn use_after_free() {
    let ptr = Box::new(42);  // Allocating memory
    let raw_ptr = Box::into_raw(ptr);  // Convert Box to raw pointer

    // Free the memory without using unsafe
    // This will not compile because converting raw pointer back to Box is unsafe
    Box::from_raw(raw_ptr);

    // Accessing freed memory without using unsafe
    // This will not compile because dereferencing a raw pointer is unsafe
    let value = *raw_ptr;

    println!("Use-After-Free executed");
}

fn main() {
    use_after_free();
}